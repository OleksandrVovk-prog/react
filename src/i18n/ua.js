export default {
  welcome: 'Спробуємо створити щось неперевершене',
  home: 'Головна',
  about: 'Про нас',
  lazy: 'Lazy',
  memo: 'Memo',
  context: 'Context',
  aboutPage: 'Про нас',
  homePage: 'Головна сторінка',
  newJoke: 'Новий жарт',
  lazyPage: 'React lazy сторінка',
  search: 'Пошук',
  title: 'Назва:',
  production: 'Виробник:',
  likes: 'Лайки:',
  like: 'Лайк'
};
