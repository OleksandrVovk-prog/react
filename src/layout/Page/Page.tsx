import { Outlet } from 'react-router-dom';
import Header from '../Header/Header';

function Page(): JSX.Element {
  return (
    <div className="page">
      <Header />
      <Outlet />
    </div>
  );
}

export default Page;
