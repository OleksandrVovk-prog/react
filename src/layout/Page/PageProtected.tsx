import { Navigate } from 'react-router-dom';
import { useAppSelector } from '../../store/hooks/useApp';
import { selectUserId } from '../../store/slices/auth/selectors';
import Page from './Page';

import IPageProtected from './interfaces/IPageProtected';

/**
 * Protected page layout
 */
function PageProtected({ redirectUri }: IPageProtected): JSX.Element {
  const id = useAppSelector(selectUserId);
  return id ? (
    <Page />
  ) : (
    <Navigate to={redirectUri} />
  );
}

export default PageProtected;
