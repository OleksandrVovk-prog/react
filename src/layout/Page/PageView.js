import { Outlet } from 'react-router-dom';
import Header from '../Header/Header';
import styles from './sass/Page.module.scss';

export default function PageView() {
  return (
    <div className={styles.app}>
      <Header />
      <Outlet />
    </div>
  );
}
