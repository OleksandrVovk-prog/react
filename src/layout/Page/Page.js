import { useCallback, useMemo } from 'react';
import i18n, { appLocales } from '../../i18n';
import PageView from './PageView';
import {
  Context as PageContext,
  data
} from '../../context/layout/Page';

export default function Page() {
  const { i18nConfig } = data;

  const changeLanguage = useCallback(() => {
    const nextLocaleIndex = appLocales.indexOf(i18n.language) + 1;
    const locale = nextLocaleIndex >= appLocales.length ? appLocales[0] : appLocales[nextLocaleIndex];
    i18n.changeLanguage(locale);
  }, []);

  const pageContextValue = useMemo(() => ({ changeLanguage, i18nConfig }), [i18nConfig, changeLanguage]);

  return (
    <PageContext.Provider value={pageContextValue}>
      <PageView />
    </PageContext.Provider>
  );
}
