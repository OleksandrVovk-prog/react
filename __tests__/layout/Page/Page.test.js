import React from 'react';
import userEvent from '@testing-library/user-event';
import i18n from 'i18next';
import { render } from '../../settings/test-utils';
import Page from '../../../src/layout/Page/Page';

describe('<Page />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(
      <Page />
    );
  });

  test('Page component should render', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should render button', () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('toggle-language-button');
    expect(Button).toBeInTheDocument();
  });

  test('should change language', async () => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('toggle-language-button');
    expect(Button).toHaveTextContent(i18n.language);
    await userEvent.click(Button);
    expect(Button).toHaveTextContent(i18n.language);
  });

  test('should render about', () => {
    const { getByText } = wrapper;
    expect(getByText(i18n.t('about'))).toBeInTheDocument();
  });

  test('should render lazy', () => {
    const { getByText } = wrapper;
    expect(getByText(i18n.t('lazy'))).toBeInTheDocument();
  });
});
