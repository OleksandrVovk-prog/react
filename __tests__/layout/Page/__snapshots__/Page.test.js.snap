// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Page component should render 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="appHeader"
    >
      <h1
        class="appTitle"
      >
        Try to build something great
      </h1>
      <a
        class="headerLink"
        href="/"
      >
        Home
      </a>
      <a
        class="headerLink"
        href="/about"
      >
        About
      </a>
      <a
        class="headerLink"
        href="/lazy"
      >
        Lazy
      </a>
      <a
        class="headerLink"
        href="/memo"
      >
        Memo
      </a>
      <a
        class="headerLink"
        href="/context"
      >
        Context
      </a>
      <button
        class="toggleLanguage"
        data-testid="toggle-language-button"
        type="button"
      >
        en
      </button>
    </header>
  </div>
</DocumentFragment>
`;
